src/main/java/fi/hsl/transitdata/tripupdate/gtfsrt/GtfsRtFactory.java

```diff
@@ -18,63 +18,75 @@
     private GtfsRtFactory() {
     }
 
-    public static GtfsRealtime.TripUpdate.StopTimeUpdate newStopTimeUpdate(StopEvent stopEvent) {
-        return newStopTimeUpdateFromPrevious(stopEvent, null);
+    public static GtfsRealtime.TripUpdate.StopTimeUpdate newStopTimeUpdate(InternalMessages.StopEstimate stopEstimate) {
+        return newStopTimeUpdateFromPrevious(stopEstimate, null);
     }
 
-    public static GtfsRealtime.TripUpdate.StopTimeUpdate newStopTimeUpdateFromPrevious(StopEvent stopEvent, GtfsRealtime.TripUpdate.StopTimeUpdate previousUpdate) {
+    public static GtfsRealtime.TripUpdate.StopTimeUpdate newStopTimeUpdateFromPrevious(
+            final InternalMessages.StopEstimate stopEstimate,
+            GtfsRealtime.TripUpdate.StopTimeUpdate previousUpdate) {
 
         GtfsRealtime.TripUpdate.StopTimeUpdate.Builder stopTimeUpdateBuilder = null;
         if (previousUpdate != null) {
             stopTimeUpdateBuilder = previousUpdate.toBuilder();
         } else {
+            String stopId = stopEstimate.getStopId();
+            int stopSequence = stopEstimate.getStopSequence();
             stopTimeUpdateBuilder = GtfsRealtime.TripUpdate.StopTimeUpdate.newBuilder()
-                    .setStopId(String.valueOf(stopEvent.getRouteData().getStopId()))
-                    .setStopSequence(stopEvent.getStopSeq());
+                    .setStopId(stopId)
+                    .setStopSequence(stopSequence);
         }
 
-        switch (stopEvent.getScheduleRelationship()) {
-            case Skipped:
+        switch (stopEstimate.getStatus()) {
+            case SKIPPED:
                 stopTimeUpdateBuilder.setScheduleRelationship(GtfsRealtime.TripUpdate.StopTimeUpdate.ScheduleRelationship.SKIPPED);
                 break;
-            case Scheduled:
+            case SCHEDULED:
                 stopTimeUpdateBuilder.setScheduleRelationship(GtfsRealtime.TripUpdate.StopTimeUpdate.ScheduleRelationship.SCHEDULED);
                 break;
         }
+        // GTFS-RT treats times in seconds
+        long stopEventTimeInSeconds = stopEstimate.getEstimatedTimeUtcMs() / 1000;
```

Comment:
Scary, need to keep this conversion in mind.

Reply:
ours always in ms, public in secs